src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs

```diff
@@ -282,7 +282,7 @@ private ICrossChainTransfer[] ValidateCrossChainTransfers(ICrossChainTransfer[]
                     // Remove any remnants of suspended transactions from the wallet.
                     foreach (KeyValuePair<ICrossChainTransfer, CrossChainTransferStatus?> kv in tracker)
                     {
-                        if (kv.Value == CrossChainTransferStatus.Suspended)
+                        if (kv.Key.Status == CrossChainTransferStatus.Suspended)
```

Comment:
No use in looking at the old status.

Reply:
This change doesn't make sense to me - the tracker sets the status both as the value and on the object.

Reply:
The tracker holds the old status and the object holds the new status. So we need to look at `kv.Key` which is the object.